forge-ai/src/main/java/forge/ai/ability/CountersPutAi.java

```diff
@@ -877,6 +877,18 @@ protected boolean doTriggerAINoCost(Player ai, SpellAbility sa, boolean mandator
                 }
             }
         }
+
+        // TODO: Unify the AI logic checking code elsewhere so that it can be checked similarly
+        // for AB, DB, or trigger PutCounter alike.
+        String logic = sa.getParam("AILogic");
```

Comment:
You should move this at the beginning of the relevant targeting if (L791)
Because the trigger is mandatory, so even if there's no favorable Fight (false returned) you still need to target (and at least getting a counter out of it is obviously also desirable)

Reply:
:+1: Good call! Done :)

Reply:
You can only return if canFight() is true (which indicates targeting was successful)

Reply:
Ah, true :)